### ISSUE TITLE
no-inner-declarations should permit inner let and const

### ISSUE THREAD

**Issue by btmills** [eslint/eslint#1893]
```js
// test.js
if (foo) {
    let bar = 42;
    const baz = bar;
}
```

```js
// config.json
{
    "rules": {
        "no-inner-declarations": [2, "both"]
    },
    "ecmaFeatures": {
        "blockBindings": true
    }
}
```

```sh
$ eslint --reset --no-eslintrc -c config.json test.js

test.js
  3:4  error  Move variable declaration to program root  no-inner-declarations
  4:4  error  Move variable declaration to program root  no-inner-declarations

✖ 2 problem (2 errors, 0 warnings)
```

Since `let` and `const` are scoped within the `if`'s consequent, moving them to the program root would functionally change the code. `no-inner-declarations` should check the `kind` of the `VariableDeclaration` node and only warn when `kind` is `"var"`.

Even with the `no-var` rule enabled, `no-inner-declarations` can still be used to prevent inner function declarations, so it would not be sufficient to disable `no-inner-declarations` when using block bindings.
